packages/components/src/custom-select-control/index.tsx

```diff
@@ -154,11 +154,11 @@ function CustomSelectControl< T extends CustomSelectOption >(
 	const renderSelectedValueHint = () => {
 		const selectedOptionHint = options
 			?.map( applyOptionDeprecations )
-			?.find( ( { name } ) => currentValue === name )?.hint;
+			?.find( ( { name } ) => value?.name === name )?.hint;
```

Comment:
This is not "_prioritizing_ the value from the props", it's _replacing_ the value from store with the value from props. The failing unit test tells us that it's a breaking change.

The problem has to do with stale state, due to this component (`CustomSelectControl`) not re-rendering after a selection change when in uncontrolled mode. Since `renderSelectedValueHint` does get re-executed when `_CustomSelect` itself re-renders, we can change the code to get fresh state on each call:

```suggestion
			?.find( ( { name } ) => store.getState().value === name )?.hint;
```